### PR TITLE
whatmp3: init at 3.8

### PR DESCRIPTION
**Pull request by dominikh**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/whatmp3/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  postInstall = ''
+    wrapProgram $out/bin/whatmp3 \
+      --prefix PATH ":" ${flac}/bin \
+      --prefix PATH ":" ${lame}/bin \
+      --prefix PATH ":" ${vorbis-tools}/bin \
+      --prefix PATH ":" ${vorbisgain}/bin \
+      --prefix PATH ":" ${mktorrent}/bin \
+      --prefix PATH ":" ${mp3gain}/bin \
+      --prefix PATH ":" ${aacgain}/bin \
+      --prefix PATH ":" ${sox}/bin
```
Comment (jtojnar) [inline pkgs/applications/audio/whatmp3/default.nix:29]: You can use

```suggestion
      --prefix PATH ":" "${stdenv.lib.makeBinPath [
        flac
        lame
        vorbis-tools
        vorbisgain
        mktorrent
        mp3gain
        aacgain
        sox
      ]}"
```

File: pkgs/applications/audio/whatmp3/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    owner = "RecursiveForest";
+    repo = "whatmp3";
+    sha256 = "0vqczxdk90sdld0yxgc1kknlyszmb4ys1bhr53n39nbhgwssh1df";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  postPatch = ''
+    substituteInPlace config.mk --replace /usr/local $out
+    substituteInPlace whatmp3.py --replace "#!/usr/bin/env python3" "#!${python3}/bin/python3"
```
Comment (jtojnar) [inline pkgs/applications/audio/whatmp3/default.nix:17]: This should be automatic if you add `python3` to `buildInputs`.

File: pkgs/applications/audio/whatmp3/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      --prefix PATH ":" ${aacgain}/bin \
+      --prefix PATH ":" ${sox}/bin
+  '';
+
+  meta = with stdenv.lib; {
+    description = "whatmp3 transcodes audio files and creates torrents for them";
+    homepage = "https://github.com/RecursiveForest/whatmp3";
+    license = licenses.mit;
+    maintainers = with maintainers; [ dominikh ];
+  };
```
Comment (jtojnar) [inline pkgs/applications/audio/whatmp3/default.nix:37]: Missing platforms.

**Comment by dominikh** [comment]
Good suggestions, thank you.

**Review by mdlayher — APPROVED** [review]
LGTM!

```
✔ ~/src/github.com/nixos/nixpkgs [master|✔] 
11:50 $ nixpkgs-review pr 92296
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/92296/head:refs/nixpkgs-review/1
remote: Enumerating objects: 6, done.
remote: Counting objects: 100% (6/6), done.
remote: Total 9 (delta 6), reused 6 (delta 6), pack-reused 3
Unpacking objects: 100% (9/9), 1.21 KiB | 51.00 KiB/s, done.
From ssh://github.com/NixOS/nixpkgs
   a54cd1cace0..56619d6ebb7  master               -> refs/nixpkgs-review/0
 + bdd65f3f69d...6329518af37 refs/pull/92296/head -> refs/nixpkgs-review/1  (forced update)
$ git worktree add /home/matt/.cache/nixpkgs-review/pr-92296/nixpkgs 56619d6ebb7bef4f379fd909d6909d93b3121131
Preparing worktree (detached HEAD 56619d6ebb7)
HEAD is now at 56619d6ebb7 Merge pull request #85992 from symphorien/ibus-db
$ git merge --no-commit 6329518af37151d6ba3a6726ac8b25262e513f9c
Auto-merging pkgs/top-level/all-packages.nix
Automatic merge went well; stopped before committing as requested
$ nix build --no-link --keep-going --option build-use-sandbox relaxed -f /home/matt/.cache/nixpkgs-review/pr-92296/build.nix
[3 built, 115 copied (359.8 MiB), 80.8 MiB DL]
https://github.com/NixOS/nixpkgs/pull/92296
1 package built:
whatmp3

$ nix-shell /home/matt/.cache/nixpkgs-review/pr-92296/shell.nix
these paths will be fetched (1.51 MiB download, 8.27 MiB unpacked):
  /nix/store/a4zs0slb4fmmr3js1ik2ckl282515k62-bash-interactive-4.4-p23-doc
  /nix/store/csdsyicqjal4viixw4ispx2hi17r0pz5-readline-7.0p5
  /nix/store/fdc7zi9sk8jlmw0rshcb585xawxb2r5m-bash-interactive-4.4-p23-dev
  /nix/store/k0bldh4a1blg50psfjfkmmsap6zkx9d1-bash-interactive-4.4-p23-man
  /nix/store/x4gd806afbgx0ag1jf8y7blzrrhiyx8q-bash-interactive-4.4-p23
  /nix/store/x718i4nnyhz46hrgmhb3xq6sijaq845x-bash-interactive-4.4-p23-info
copying path '/nix/store/a4zs0slb4fmmr3js1ik2ckl282515k62-bash-interactive-4.4-p23-doc' from 'https://cache.nixos.org'...
copying path '/nix/store/x718i4nnyhz46hrgmhb3xq6sijaq845x-bash-interactive-4.4-p23-info' from 'https://cache.nixos.org'...
copying path '/nix/store/k0bldh4a1blg50psfjfkmmsap6zkx9d1-bash-interactive-4.4-p23-man' from 'https://cache.nixos.org'...
copying path '/nix/store/csdsyicqjal4viixw4ispx2hi17r0pz5-readline-7.0p5' from 'https://cache.nixos.org'...
copying path '/nix/store/x4gd806afbgx0ag1jf8y7blzrrhiyx8q-bash-interactive-4.4-p23' from 'https://cache.nixos.org'...
copying path '/nix/store/fdc7zi9sk8jlmw0rshcb585xawxb2r5m-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...

[nix-shell:~/.cache/nixpkgs-review/pr-92296]$ ./results/whatmp3/bin/whatmp3 --version
.whatmp3-wrapped 3.8
```

File: pkgs/applications/audio/whatmp3/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchFromGitHub, flac, lame, vorbis-tools, vorbisgain, mktorrent, makeWrapper, python3, mp3gain, aacgain, sox }:
+
+stdenv.mkDerivation rec {
+  pname = "whatmp3";
+  version = "3.8";
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/whatmp3/default.nix:6]:
```suggestion
  version = "3.8";

  src = fetchFromGitHub {
```

File: pkgs/applications/audio/whatmp3/default.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    rev = "v3.8";
+    owner = "RecursiveForest";
+    repo = "whatmp3";
+    sha256 = "0vqczxdk90sdld0yxgc1kknlyszmb4ys1bhr53n39nbhgwssh1df";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/whatmp3/default.nix:10]:
```suggestion
    owner = "RecursiveForest";
    repo = "whatmp3";
    rev = "v${version}";
    sha256 = "0vqczxdk90sdld0yxgc1kknlyszmb4ys1bhr53n39nbhgwssh1df";
```

File: pkgs/applications/audio/whatmp3/default.nix [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        vorbisgain
+        mktorrent
+        mp3gain
+        aacgain
+        sox
+      ]}"
+  '';
+
+  meta = with stdenv.lib; {
+    description = "whatmp3 transcodes audio files and creates torrents for them";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/whatmp3/default.nix:35]:
```suggestion
    description = "Transcodes audio files and creates torrents of them";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 92296` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>whatmp3</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 92296` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>whatmp3</li>
  </ul>
</details>

**Comment by dominikh** [comment]
Sorry, I no longer think this is worth adding to nixpkgs.

**Pull request closed without merging**